include/alpaka/vec/Vec.hpp

```diff
@@ -511,8 +512,7 @@ namespace alpaka
         }
 
     private:
-        // 16 Byte alignment for usage inside of CUDA kernels.
-        alignas(16u) TSize m_data[TDim::value];
+        alignas(ALPAKA_OPTIMAL_ALIGNMENT_SIZE(TSize[TDim::value])) TSize m_data[TDim::value];
```

Comment:
**Version 1:** `alignas(ALPAKA_OPTIMAL_ALIGNMENT(TSize)) TSize m_data[TDim::value]; ` create the following ptx code
```C++
///bigdata/hplsim/scratch/widera/alpaka/include/alpaka/workdiv/WorkDivMembers.hpp:77             {}
	.loc 6 77 1
	ld.u32 	%r1, [%rd2];
	ld.u32 	%r2, [%rd2+4];
	ld.u32 	%r3, [%rd2+8];
	st.u32 	[%rd1+8], %r3;
	st.u32 	[%rd1+4], %r2;
	st.u32 	[%rd1], %r1;
	ld.u32 	%r4, [%rd2+12];
	ld.u32 	%r5, [%rd2+16];
	ld.u32 	%r6, [%rd2+20];
	st.u32 	[%rd1+20], %r6;
	st.u32 	[%rd1+16], %r5;
	st.u32 	[%rd1+12], %r4;
	ld.u32 	%r7, [%rd2+24];
	ld.u32 	%r8, [%rd2+28];
	ld.u32 	%r9, [%rd2+32];
	st.u32 	[%rd1+32], %r9;
	st.u32 	[%rd1+28], %r8;
	st.u32 	[%rd1+24], %r7;
```
**Version 2:** `alignas(ALPAKA_OPTIMAL_ALIGNMENT(TSize[TDim::value])) TSize m_data[TDim::value]; ` create the following ptx code
```C++
///bigdata/hplsim/scratch/widera/alpaka/include/alpaka/workdiv/WorkDivMembers.hpp:77             {}
	.loc 7 77 1
	ld.v4.u32 	{%r1, %r2, %r3, %r4}, [%rd2];
	ld.v4.u8 	{%rs1, %rs2, %rs3, %rs4}, [%rd2+12];
	ld.v2.u32 	{%r6, %r7}, [%rd2+16];
	ld.v4.u8 	{%rs9, %rs10, %rs11, %rs12}, [%rd2+28];
	st.v4.u8 	[%rd1+12], {%rs1, %rs2, %rs3, %rs4};
	st.v2.u32 	[%rd1], {%r1, %r2};
	st.u32 	[%rd1+8], %r3;
	st.v2.u8 	[%rd1+28], {%rs9, %rs10};
	st.v2.u32 	[%rd1+16], {%r6, %r7};
	st.u8 	[%rd1+30], %rs11;
	ld.u32 	%r12, [%rd2+24];
	ld.u8 	%rs16, [%rd2+31];
	st.u8 	[%rd1+31], %rs16;
	st.u32 	[%rd1+24], %r12;
	ld.u32 	%r13, [%rd2+32];
	ld.u32 	%r14, [%rd2+36];
	ld.u32 	%r15, [%rd2+40];
	ld.u8 	%rs17, [%rd2+44];
	ld.u8 	%rs18, [%rd2+45];
	ld.u8 	%rs19, [%rd2+46];
	ld.u8 	%rs20, [%rd2+47];
	st.u8 	[%rd1+47], %rs20;
	st.u8 	[%rd1+46], %rs19;
	st.u8 	[%rd1+45], %rs18;
	st.u8 	[%rd1+44], %rs17;
	st.u32 	[%rd1+40], %r15;
	st.u32 	[%rd1+36], %r14;
	st.u32 	[%rd1+32], %r13;
```

Because of the alignment in the second version the compiler can put the data in registers and must copy all byte by byte.